### ISSUE TITLE
Craft 4 [request]

### ISSUE THREAD

**Issue by magicspon** [jamesedmonston/graphql-authentication#94]
Hi James,

Are you planning on updating this plugin to work with Craft 4?  If so, do you have any idea when it might be ready?

ta

**Comment by mgambella** [comment]
Hello @jamesedmonston have you some news on this issue? When do you think you will release the updated plugin?

**Comment by blnkt** [comment]
Such excellent news.  Look forward to a Craft 4 compatible release.

**Comment by howells** [comment]
@jamesedmonston so sorry to chase but do you know when the Craft 4 version might be ready? This is the only plugin preventing migration one one site that desperately needs the multisite performance improvements of 4 😅

**Comment by robinbeatty** [comment]
+1 for Craft 4 version. Any update on headway? Thanks

**Comment by howells** [comment]
@brandonkelly is there any chance of rolling this functionality natively into Craft 4? :-)

**Comment by brandonkelly** [comment]
@howells We’re looking into it, but probably more of a 5.0 thing, not 4.x. See craftcms/cms#8757 for a related discussion

**Comment by robinbeatty** [comment]
@jamesedmonston How big a job do you reckon it is to get the plugin in shape for Craft 4? I've just got to this requirement for a project and hadn't actually realised it's not a native feature...

**Comment by brandonkelly** [comment]
Just went ahead and ported the plugin to Craft 4 from a personal fork ([`brandonkelly:craft4`](https://github.com/brandonkelly/graphql-authentication/tree/craft4)).

It’s branched off of [`brandonkelly:develop`](https://github.com/brandonkelly/graphql-authentication/tree/develop) which is a few commits ahead of [`jamesedmonston:develop`](https://github.com/jamesedmonston/graphql-authentication/tree/develop) (PR #98), so will wait until that’s merged before submitting a new PR for Craft 4.

You can pull my `craft4` branch in by making these changes to `composer.json`:

```json
"require": {
  "...": "...",
  "jamesedmonston/graphql-authentication": "craft4-dev"
},
"repositories": [
  {"type": "git", "url": "https://github.com/brandonkelly/graphql-authentication"}
]
```

If you run into any issues specifically with the Craft 4 port, please report those over at https://github.com/brandonkelly/graphql-authentication/issues.

**Comment by Stalex89** [comment]
Hello @brandonkelly, thank you for a temporary fix for the plugin, however I'm still running into the issue when trying to install the plugin.

When I'm using the code from your post above, it says the following:

![Screenshot 2022-08-31 at 17 11 00](https://user-images.githubusercontent.com/23526237/187714658-76e33db4-5113-44e4-b941-a7150ad435f5.png)

when I change from `  "jamesedmonston/graphql-authentication": "craft4-dev"` to `  "jamesedmonston/graphql-authentication": "dev-develop"`, I'm getting the following:

![Screenshot 2022-08-31 at 17 11 26](https://user-images.githubusercontent.com/23526237/187714936-24afb60b-556f-4015-967b-45bc2ca85f7d.png)

My composer.json: 

```
{
  "require": {
    "craftcms/cms": "4.2.3",
    "craftcms/redactor": "3.0.2",
    "nystudio107/craft-seomatic": "4.0.7",
    "sebastianlenz/linkfield": "2.1.4",
    "verbb/super-table": "3.0.0",
    "vlucas/phpdotenv": "^5.4.0",
    "jamesedmonston/graphql-authentication": "dev-develop"
  },
  "require-dev": {
    "yiisoft/yii2-shell": "^2.0.3"
  },
  "repositories": [
    {"type": "git", "url": "https://github.com/brandonkelly/graphql-authentication"}
  ],
  "autoload": {
    "psr-4": {
      "modules\\": "modules/"
    }
  },
  "config": {
    "allow-plugins": {
      "craftcms/plugin-installer": true,
      "yiisoft/yii2-composer": true
    },
    "sort-packages": true,
    "optimize-autoloader": true,
    "platform": {
      "php": "8.0.2"
    }
  },
  "scripts": {
    "post-root-package-install": [
      "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
    ]
  }
}
```

Can you pls tell me if I'm missing something ? Thank you in advance

**Comment by Stalex89** [comment]
@robinbeatty Hello, can you please show me your `composer.json` content? I'm getting this error:

![Screenshot 2022-08-31 at 17 25 43](https://user-images.githubusercontent.com/23526237/187717634-f6804fdc-7615-48f3-b925-48ddde3b3fb6.png)

**Comment by robinbeatty** [comment]
Hey Alex 
<img width="652" alt="Screenshot 2022-08-31 at 16 30 32" src="https://user-images.githubusercontent.com/48855073/187718375-84eeec81-a4e1-46b1-a82d-44c6ac490305.png">


Just spotted the latest merge of @brandonkelly's from @jamesedmonston so maybe this has changed?

**Comment by brandonkelly** [comment]
Yeah the Craft 4 changes are now merged into `jamesedmonston:develop` so I’ve deleted my `craft4` branch. The custom `repositories` config should not be needed anymore, and the `jamesedmonston/graphql-authentication` constraint can be set to `dev-develop`.

```json
"require": {
  "...": "...",
  "jamesedmonston/graphql-authentication": "dev-develop"
}
```

**Comment by robinbeatty** [comment]
Cheers for clarification :)

**Comment by Stalex89** [comment]
Works like charm now, thank you for clarification! 😀

**Comment by jamesedmonston** [comment]
Hi everyone,

I'm currently working on fixing a few longstanding issues/feature requests and will be pushing out a release shortly (hopefully this evening).

Huge thanks to @brandonkelly for taking the time to make the Craft 4 updates. 💪

**Comment by jamesedmonston** [comment]
I've just released 2.0 which has @brandonkelly's updates included – thanks again @brandonkelly.

If anybody finds any issues please can they be created as new issues?

Thanks!
